exonum-java-binding/core/rust/integration_tests/tests/runtime.rs

```diff
@@ -0,0 +1,667 @@
+    fake_service::{
+        self, create_before_commit_test_map, create_init_service_test_map,
+        create_service_artifact_non_instantiable_service, create_service_artifact_non_loadable,
+        create_service_artifact_valid, create_tx_test_entry,
+    },
+    vm::{create_vm_for_tests_with_fake_classes, log4j_path},
+};
+use java_bindings::{
+    create_service_runtime,
+    exonum::{
+        blockchain::{Blockchain, ExecutionError, ExecutionErrorKind, InstanceConfig},
+        crypto::{gen_keypair, hash, Hash, PublicKey, SecretKey},
+        messages::{AnyTx, Verified},
+        node::ApiSender,
+        runtime::{ArtifactId, CallInfo, InstanceId, InstanceSpec, MethodId, Runtime},
+    },
+    exonum_merkledb::{BinaryValue, TemporaryDB},
+    jni::JavaVM,
+    utils::any_to_string,
+    DeployArguments, Error, Executor, JavaRuntimeProxy, RuntimeConfig,
+};
+use std::{panic, path::PathBuf, sync::Arc};
+use tempfile::TempPath;
+use tokio_core::reactor::Core;
+
+lazy_static! {
+    static ref VM: Arc<JavaVM> = create_vm_for_tests_with_fake_classes();
+}
+
+const VALID_INSTANCE_ID: u32 = 127;
+const VALID_INSTANCE_NAME: &str = "artifact__test_valid";
+const VALID_ARTIFACT_NAME: &str = "artifact:test-valid";
+const VALID_ARTIFACT_VERSION: &str = "1.0";
+
+#[test]
+fn runtime_exception_handling_checked_exception() {
+    let artifact = ArtifactId::new(JavaRuntimeProxy::RUNTIME_ID, "artifact:test-name:1.0").unwrap();
+    let mut runtime = get_fake_runtime("createRuntimeThrowingExceptions");
+    let future = runtime.deploy_artifact(artifact, vec![]);
+
+    let mut core = Core::new().unwrap();
+    let result = core.run(future);
+
+    assert!(result.is_err());
+
+    let error = result.unwrap_err();
+    let message = error.description;
+    let kind = error.kind;
+
+    match kind {
+        ExecutionErrorKind::Runtime { code } => {
+            assert_eq!(code, Error::JavaException as u8);
+            assert!(message.contains("deployArtifact"));
+            assert!(message.contains("ServiceLoadingException"))
+        }
+        _ => panic!("error.kind is not 'ExecutionErrorKind::Runtime'"),
+    }
+
+    runtime.shutdown();
+}
+
+#[test]
+fn runtime_exception_handling_unchecked_exception() {
+    let artifact = ArtifactId::new(JavaRuntimeProxy::RUNTIME_ID, "artifact:test-name:1.0").unwrap();
+    let mut runtime = get_fake_runtime("createRuntimeThrowingExceptions");
+    assert_panics("isArtifactDeployed", || {
+        runtime.is_artifact_deployed(&artifact)
+    });
+    runtime.shutdown();
+}
+
+#[test]
+fn runtime_initialize_and_shutdown() {
+    let mut runtime = create_runtime();
+    let blockchain = create_blockchain();
+
+    runtime.initialize(&blockchain);
+    runtime.shutdown();
+}
+
+#[test]
+fn load_artifact() {
+    let mut runtime = create_runtime();
+    let mut core = Core::new().unwrap();
+    let blockchain = create_blockchain();
+
+    let artifact = create_artifact(
```

Comment:
(1) move closer to where it is used (same to blockchain).
 (2) clarify the steps